### PR TITLE
Publish release binaries and docker images

### PR DESCRIPTION
**Pull request by dyasny**
Github action that publishes the sqld release binaries for linux and macosx x86

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,16 +1,17 @@
-name: Create and publish a Docker image
+name: Create and publish qld binaried and Docker image
```
Comment (MarinPostma) [inline .github/workflows/publish.yml:1]:
```suggestion
name: Create and publish sqld binaries and Docker image
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -43,3 +44,174 @@ jobs:
...
+      - uses: actions/checkout@v3
+
+      # Check Cargo.toml in sqld sbdir has the correct tag
+      - name: Verify sqld/Cargo.toml version is same as git tag
+        if: ${{ startsWith(github.ref, 'refs/tags/') }} && ${{ matrix.os }} == ubuntu-latest
+        id: check-tag
+        run: |
+          VER=$(echo ${{ github.ref_name }}|tr -d "v")
+          pushd sqld
+          CARGOVER=$(cargo metadata --no-deps | jq '.packages | .[] | select(.name == "sqld") | .version'|tr -d '"')
```
Comment (MarinPostma) [inline .github/workflows/publish.yml:72]: will this work before you install a toolchain?

Comment (MarinPostma) [reply .github/workflows/publish.yml:72]: also, is jq already installed?

Comment (dyasny) [reply .github/workflows/publish.yml:72]: jq is there (see https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md) but the order needs to be changed, I'll fix it

**Review by MarinPostma — APPROVED** [review]

**Pull request merged**